File: sandbox-runner/patterns/saga/python/starter.py
import asyncio
import time

from temporalio.client import Client

from shared import TASK_QUEUE, WORKFLOW_ID_PREFIX, OpenAccountRequest
from workflows import OpenAccountWorkflow


async def main() -> None:
    client = await Client.connect("localhost:7233")
    account_id = f"{WORKFLOW_ID_PREFIX}-{int(time.time() * 1000)}"
    request = OpenAccountRequest(
        accountId=account_id,
        clientName="Alice Example",
        clientEmail="alice@example.com",
        address="123 Main St, Brooklyn NY",
        bankAccount="DE89-3704-0044-0532-0130-00",
    )

    handle = await client.start_workflow(
        OpenAccountWorkflow.run,
        request,
        id=account_id,
        task_queue=TASK_QUEUE,
    )
    print(f"Started workflow: {account_id}")

    result = await handle.result()
    print(result)
    print(f"Open the Temporal UI and search for '{account_id}' to see the saga history.")


if __name__ == "__main__":
    asyncio.run(main())

File: sandbox-runner/patterns/saga/java/src/main/java/Starter.java
import io.temporal.client.WorkflowClient;
import io.temporal.client.WorkflowOptions;
import io.temporal.serviceclient.WorkflowServiceStubs;

public class Starter {
    public static void main(String[] args) {
        WorkflowServiceStubs service = WorkflowServiceStubs.newLocalServiceStubs();
        WorkflowClient client = WorkflowClient.newInstance(service);

        String accountId = Shared.WORKFLOW_ID_PREFIX + "-" + System.currentTimeMillis();
        Shared.OpenAccountRequest req = new Shared.OpenAccountRequest(
                accountId,
                "Alice Example",
                "alice@example.com",
                "123 Main St, Brooklyn NY",
                "DE89-3704-0044-0532-0130-00");

        OpenAccountWorkflow workflow = client.newWorkflowStub(
                OpenAccountWorkflow.class,
                WorkflowOptions.newBuilder()
                        .setTaskQueue(Shared.TASK_QUEUE)
                        .setWorkflowId(accountId)
                        .build());

        System.out.println("Started workflow: " + accountId);

        String result = workflow.openAccount(req);
        System.out.println(result);
        System.out.println("Open the Temporal UI and search for '" + accountId
                + "' to see the saga history.");
    }
}
